### ISSUE TITLE
Create dataset loader for MedHop

### ISSUE THREAD

**Issue by jason-fries** [bigscience-workshop/biomedical#225]
## Adding a Dataset
- **Name:** MedHop
- **Description:** *None provided*
- **Task:** QA
- **Paper:** https://transacl.org/ojs/index.php/tacl/article/viewFile/1325/299
- **Data:** http://qangaroo.cs.ucl.ac.uk
- **License:** CC BY-SA 3.0

**Comment by qanastek** [comment]
#self-assign

**Comment by qanastek** [comment]
I migrate the project to [Zenodo](https://zenodo.org/record/6407402#.Ykecq-dBybi) since the shortcoming Google Drive is that the integrated antivirus system cannot scan files that are larger than 100 MB and the corpus is 324 MB. After this 100 MB barrier, downloading the file require a manual step to accept the "conditions" by clicking on the `Accept` button :

```plain
(324M) is too large for Google to scan for viruses. Would you still like to download this file?
```

The Zenodo repository contains the exact `.zip` file provided by the Google Drive since the [Creative Commons Attribution Share Alike 3.0 Unported](https://creativecommons.org/licenses/by-sa/3.0/) license allow any copy and redistribution.

**Comment by leonweber** [comment]
I think you should be able to download the dataset programatically via [gdown](https://pypi.org/project/gdown/)

**Comment by qanastek** [comment]
I have not use [gdown](https://pypi.org/project/gdown/) since it's not available in the requirements of the project or the [datasets](https://github.com/huggingface/datasets) library. Or either, should I make a check inside my code and ask to the user the installation of it ? Like this :

````python
import importlib.util

if importlib.util.find_spec("gdown") is None:
    print("The package \033[93mgdown\033[0m is not installed! \033[0mPlease use the \033[93mpip install gdown\033[0m command.")
    exit(1)
else:
    import gdown
    # Do my stuff here
````

**Comment by galtay** [comment]
Having examine a few other corner cases, I think we are converging on "If the `dl_manager` object provided by datasets can not easily download the dataset, then implement is as local.

**Comment by galtay** [comment]
@qanastek update from https://github.com/lhoestq  ... the `dl_manager` does support google drive (with the google drive direct download link). can you try an implement as in this example? https://huggingface.co/datasets/cnn_dailymail/blob/main/cnn_dailymail.py#L67

**Comment by qanastek** [comment]
@galtay I already tried it :

I use this code :

```python
# ORIGNAL_URL = "https://drive.google.com/u/0/uc?id=1ytVZ4AhubFDOEL7o7XrIRIyhU8g9wvKA&export=download"
# "cnn_stories": "https://drive.google.com/uc?export=download&id=0BwmD_VLjROrfTHk4NFg2SndKcjQ"
URL_GDRIVE     = "https://drive.google.com/uc?export=download&id=1ytVZ4AhubFDOEL7o7XrIRIyhU8g9wvKA" # Same url shape as in CNN Dailymail
dl_paths = dl_manager.download_and_extract(URL_GDRIVE)
```

When I check the single file downloaded we got using :

```plain
cat /home/home/.cache/huggingface/datasets/downloads/aec412344416349cdf4b2234eb6c3158ace0d0be04dfb2abd18c0b4ce2d6937f
```

We can see that the dl_manager had simply download the html page :

```html
<!DOCTYPE html>
<html>
   <head>
      <title>Google Drive - Virus scan warning</title>
      <meta http-equiv="content-type" content="text/html; charset=utf-8"/>
      <style nonce="AxF19dYPZRZT0LymLezVaQ">/* Copyright 2022 Google Inc. All Rights Reserved. */
         .goog-inline-block{position:relative;display:-moz-inline-box;display:inline-block}* html .goog-inline-block,*:first-child+html .goog-inline-block{display:inline}.goog-link-button{position:relative;color:#15c;text-decoration:underline;cursor:pointer}.goog-link-button-disabled{color:#ccc;text-decoration:none;cursor:default}body{color:#222;font:normal 13px/1.4 arial,sans-serif;margin:0}.grecaptcha-badge{visibility:hidden}.uc-main{padding-top:50px;text-align:center}#uc-dl-icon{display:inline-block;margin-top:16px;padding-right:1em;vertical-align:top}#uc-text{display:inline-block;max-width:68ex;text-align:left}.uc-error-caption,.uc-warning-caption{color:#222;font-size:16px}#uc-download-link{text-decoration:none}.uc-name-size a{color:#15c;text-decoration:none}.uc-name-size a:visited{color:#61c;text-decoration:none}.uc-name-size a:active{color:#d14836;text-decoration:none}.uc-footer{color:#777;font-size:11px;padding-bottom:5ex;padding-top:5ex;text-align:center}.uc-footer a{color:#15c}.uc-footer a:visited{color:#61c}.uc-footer a:active{color:#d14836}.uc-footer-divider{color:#ccc;width:100%}
      </style>
      <link rel="icon" href="null"/>
   </head>
   <body>
      <div class="uc-main">
         <div id="uc-dl-icon" class="image-container">
            <div class="drive-sprite-aux-download-file"></div>
         </div>
         <div id="uc-text">
            <p class="uc-warning-caption">Google Drive can't scan this file for viruses.</p>
            <p class="uc-warning-subcaption"><span class="uc-name-size"><a href="/open?id=1ytVZ4AhubFDOEL7o7XrIRIyhU8g9wvKA">qangaroo_v1.1.zip</a> (324M)</span> is too large for Google to scan for viruses. Would you still like to download this file?</p>
            <form id="downloadForm" action="https://drive.google.com/uc?export=download&amp;id=1ytVZ4AhubFDOEL7o7XrIRIyhU8g9wvKA&amp;export=download&amp;confirm=t" method="post"><input type="submit" id="uc-download-link" class="goog-inline-block jfk-button jfk-button-action" value="Download anyway"/></form>
         </div>
      </div>
      <div class="uc-footer">
         <hr class="uc-footer-divider">
      </div>
   </body>
</html>
```

**Comment by qanastek** [comment]
https://github.com/bigscience-workshop/biomedical/issues/225#issuecomment-1086943581 Have you any example that following the BigScience Biomedical guideline ?

**Comment by jason-fries** [comment]
@qanastek I believe this URL should work? Can you test?

```
https://drive.google.com/uc?export=download&confirm=yTib&id=1ytVZ4AhubFDOEL7o7XrIRIyhU8g9wvKA
```

**Comment by galtay** [comment]
this is working for me locally, 
```
import os
from datasets.utils.download_manager import DownloadManager


_URLS = "https://drive.google.com/uc?export=download&id=1ytVZ4AhubFDOEL7o7XrIRIyhU8g9wvKA"
dl_manager = DownloadManager()
dl_paths = dl_manager.download_and_extract(_URLS)
print(os.listdir(os.path.join(dl_paths, "qangaroo_V1.1", "medhop")))
```

can you confirm @qanastek ?

**Comment by galtay** [comment]
.... also just thinking about some potential pitfalls ... you might want to clear your datasets cache

**Comment by qanastek** [comment]
https://github.com/bigscience-workshop/biomedical/issues/225#issuecomment-1087014365 I cleared it just before the example on top.

**Comment by qanastek** [comment]
@jason-fries Your link is working perfectly! Replacing `id` with `confirm` solve the problem.

**Comment by qanastek** [comment]
Commit df06a8074865ba124d6f413b3109be0d64dc0b41

**Comment by qanastek** [comment]
PR https://github.com/bigscience-workshop/biomedical/pull/322

**Comment by hakunanatasha** [comment]
Merged with #322
